### PR TITLE
TER-66 incremental load on tables without PKs

### PR DESCRIPTION
**Pull request by jirkasemmler**
Jira: TER-66

Before asking for review make sure that:

- [x] In the case of **modified import queries**, I checked that the [CustomQuery component](https://github.com/keboola/app-custom-query-manager) is updated.

---

Incremental load without PK -> just INSERT INTO
+ trait for deleting table if exists (TD doesnt support it -> we need combination of two commands)

### CODE REVIEW

**Comment by martinjunger** [comment]
👀

File: tests/functional/Teradata/TeradataBaseTestCase.php [hunk 1]
```diff
@@ -363,6 +380,42 @@ protected function initTable(string $tableName, string $dbName = ''): void
                     TeradataQuote::quoteSingleIdentifier($this->getSourceDbName())
                 ));
                 break;
+            case self::TABLE_ACCOUNTS_BEZ_TS:
```
Comment (martinjunger) [inline tests/functional/Teradata/TeradataBaseTestCase.php:385]: `_BEZ_` -> `_WITHOUT_` ;-)

File: tests/functional/Teradata/ToFinal/IncrementalImportTest.php [hunk 2]
```diff
@@ -69,6 +69,14 @@ public function incrementalImportData(): Generator
         $multiPkColumns = array_shift($expectedMultiPkRows);
         $expectedMultiPkRows = array_values($expectedMultiPkRows);
 
+        $multiPkExpectationsWithoutPKFile =  new CsvFile(self::DATA_DIR . 'multi-pk.csv');
+        $multiPkExpectationsWithoutPKROws = [];
```
Comment (martinjunger) [inline tests/functional/Teradata/ToFinal/IncrementalImportTest.php:72]: `...ROws` -> `...Rows`

File: tests/functional/Teradata/ToFinal/IncrementalImportTest.php [hunk 3]
```diff
@@ -69,6 +69,14 @@ public function incrementalImportData(): Generator
...
+        foreach ($multiPkExpectationsWithoutPKFile as $row) {
+            $multiPkExpectationsWithoutPKROws[] = $row;
```
Comment (martinjunger) [inline tests/functional/Teradata/ToFinal/IncrementalImportTest.php:74]: `...ROws` -> `...Rows`

File: tests/functional/Teradata/ToFinal/IncrementalImportTest.php [hunk 4]
```diff
@@ -69,6 +69,14 @@ public function incrementalImportData(): Generator
...
+        }
+        // skip columnNames
+        array_shift($multiPkExpectationsWithoutPKROws);
```
Comment (martinjunger) [inline tests/functional/Teradata/ToFinal/IncrementalImportTest.php:77]: `...ROws` -> `...Rows`

File: src/Backend/Teradata/Helper/DropTableTrait.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Db\ImportExport\Backend\Teradata\Helper;
+
+use Keboola\Db\ImportExport\Backend\Teradata\ToFinalTable\SqlBuilder;
+
+trait DropTableTrait
```
Comment (martinjunger) [inline src/Backend/Teradata/Helper/DropTableTrait.php:9]: tady by se hodil jiny nazev, treba `CommandsTrait` nebo `DbCommandsTrait`...

File: src/Backend/Teradata/Helper/DropTableTrait.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public function dropIfExists($dbName, $tableName): void
```
Comment (martinjunger) [inline src/Backend/Teradata/Helper/DropTableTrait.php]: -> `dropTableIfExists`

File: phpstan-baseline.neon [hunk 7]
```diff
@@ -410,6 +405,21 @@ parameters:
 			count: 1
 			path: tests/functional/Teradata/ExportTest.php
 
+		-
+			message: "#^Parameter \\#1 \\$callback of function array_map expects \\(callable\\(array\\{Key\\: string\\}\\|Google\\\\Cloud\\\\Storage\\\\StorageObject\\|MicrosoftAzure\\\\Storage\\\\Blob\\\\Models\\\\Blob\\)\\: mixed\\)\\|null, Closure\\(Google\\\\Cloud\\\\Storage\\\\StorageObject\\)\\: string given\\.$#"
+			count: 1
+			path: tests/functional/Teradata/ToFinal/IncrementalImportTest.php
```
Comment (martinjunger) [inline phpstan-baseline.neon:411]: co tohle, da se to odstranit?

**Review by martinjunger — CHANGES_REQUESTED** [review]
Dobry, jen par veci...

File: phpstan-baseline.neon (hunk 7 above)
Comment (jirkasemmler) [reply phpstan-baseline.neon:411]: dal jsem tomu hodinu a neprisel jsem na to jak. Imho je to spis bug v phpstanu

File: tests/functional/Teradata/TeradataBaseTestCase.php (hunk 1 above)
Comment (jirkasemmler) [reply tests/functional/Teradata/TeradataBaseTestCase.php]: ty vole cely traktor!!! https://github.com/keboola/php-db-import-export/pull/151/commits/f2ca158332d13c4329cd4930930fdffb195a0d1b

File: src/Backend/Teradata/Helper/DropTableTrait.php (hunk 5 above)
Comment (jirkasemmler) [reply src/Backend/Teradata/Helper/DropTableTrait.php:9]: ty jo nevim, nemam vyhraneny nazor, muj plan s touto traitou je aby delala jen toto. nic vic, nic min, tak bych ji i tak nechal pojmenovanou. Nevidim v ostatnich traitach nejakou konvenci... 
image

Comment (martinjunger) [reply src/Backend/Teradata/Helper/DropTableTrait.php:9]: OK, az to bude potreba, tak to muzem treba prejmenovat...

**Review by martinjunger — APPROVED** [review]
OK

**Pull request merged**